Buildings/Controls/OBC/Utilities/PIDWithInputGains.mo

```diff
@@ -0,0 +1,749 @@
+within Buildings.Controls.OBC.Utilities;
+block PIDWithInputGains
+  "P, PI, PD, and PID controller with output reset and input gains"
+  parameter Buildings.Controls.OBC.CDL.Types.SimpleController controllerType=Buildings.Controls.OBC.CDL.Types.SimpleController.PI
+    "Type of controller";
+  parameter Real r(
+    min=100*CDL.Constants.eps)=1
+    "Typical range of control error, used for scaling the control error";
+  parameter Real yMax=1
+    "Upper limit of output"
+    annotation (Dialog(group="Limits"));
+  parameter Real yMin=0
+    "Lower limit of output"
+    annotation (Dialog(group="Limits"));
+  parameter Real Ni(
+    min=100*CDL.Constants.eps)=0.9
+    "Ni*Ti is time constant of anti-windup compensation"
+    annotation (Dialog(tab="Advanced",group="Integrator anti-windup",enable=controllerType == CDL.Types.SimpleController.PI or controllerType ==CDL.Types.SimpleController.PID));
+  parameter Real Nd(
+    min=100*CDL.Constants.eps)=10
+    "The higher Nd, the more ideal the derivative block"
+    annotation (Dialog(tab="Advanced",group="Derivative block",enable=controllerType == CDL.Types.SimpleController.PD or controllerType ==CDL.Types.SimpleController.PID));
+  parameter Real xi_start=0
+    "Initial value of integrator state"
+    annotation (Dialog(tab="Advanced",group="Initialization",enable=controllerType == CDL.Types.SimpleController.PI or controllerType == CDL.Types.SimpleController.PID));
+  parameter Real yd_start=0
+    "Initial value of derivative output"
+    annotation (Dialog(tab="Advanced",group="Initialization",enable=controllerType == CDL.Types.SimpleController.PD or controllerType == CDL.Types.SimpleController.PID));
+  parameter Boolean reverseActing=true
+    "Set to true for reverse acting, or false for direct acting control action";
+  parameter Real y_reset=xi_start
+    "Value to which the controller output is reset if the boolean trigger has a rising edge"
+    annotation (Dialog(enable=controllerType == CDL.Types.SimpleController.PI or controllerType == CDL.Types.SimpleController.PID,group="Integrator reset"));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_s
+    "Connector of setpoint input signal"
```

Comment:
Better write "Connector for ..." rather than "Connector of ..." (also for the other inputs)

Reply:
Thanks, I have changed "Connector of ..."  into "Connector for ..."